### ISSUE TITLE
Problem in FlowMatchEulerDiscreteScheduler's index_for_timestep method

### ISSUE THREAD

**Issue by mashrurmorshed** [huggingface/diffusers#9331]
### Describe the bug

In `index_for_timestep()` ([here](https://github.com/huggingface/diffusers/blob/d8a16635f47ac455abd61879bcc6be32dfeaa561/src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py#L213)), the provided timestep is checked against the scheduler's timesteps. However, the scheduler timesteps are floats, and it is well known that comparing floats with `==` causes problems.

```python
def index_for_timestep(self, timestep, schedule_timesteps=None):
    if schedule_timesteps is None:
        schedule_timesteps = self.timesteps

        indices = (schedule_timesteps == timestep).nonzero() # <--- HERE

        # The sigma index that is taken for the **very** first `step`
        # is always the second index (or the last index if there is only 1)
        # This way we can ensure we don't accidentally skip a sigma in
        # case we start in the middle of the denoising schedule (e.g. for image-to-image)
        pos = 1 if len(indices) > 1 else 0

        return indices[pos].item()
```

An obvious solution is to do `torch.isclose(schedule_timesteps, timestep).nonzero()` to avoid this issue. I can open a PR for this if the maintainers think this is alright.

### Reproduction

```python
from diffusers import FlowMatchEulerDiscreteScheduler

sfm = FlowMatchEulerDiscreteScheduler()
sfm.index_for_timestep(254.0, sfm.timesteps)
```

Error will not happen if I force cast to int64:

```python
sfm.index_for_timestep(254.0, sfm.timesteps.long())
```

### Logs

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda3/envs/flow-matching/lib/python3.10/site-packages/diffusers/schedulers/scheduling_flow_match_euler_discrete.py", line 225, in index_for_timestep
    return indices[pos].item()
IndexError: index 0 is out of bounds for dimension 0 with size 0
```

### System Info

🤗 Diffusers version: 0.30.1
- Platform: Linux-5.15.0-107-generic-x86_64-with-glibc2.35
- Running on Google Colab?: No
- Python version: 3.10.14
- PyTorch version (GPU?): 2.4.0+cu121 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Huggingface_hub version: 0.24.6
- Transformers version: not installed
- Accelerate version: 0.33.0
- PEFT version: not installed
- Bitsandbytes version: not installed
- Safetensors version: 0.4.4
- xFormers version: not installed
- Accelerator: NVIDIA RTX A6000, 49140 MiB
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: Yes

### Who can help?

@yiyixuxu @sayakpaul

**Comment by yiyixuxu** [comment]
hi @mashrurmorshed 
thanks for the issue! you're right about it but I would like to understand your use case before we can think about how to go about this

currently, we only use`index_for_timestep` when `begin_index` is not passed from the pipeline, and all our img2img and inpaint pipeline with flow match will pass a begin index, therefore does not need to search for the index at all, that function is there for backward compatibility

**Comment by asomoza** [comment]
closing this issue for now as there's was a month without any updates from OP so it probably the answer was enough. If there's any updates I'll reopen it or feel free to open another issue.
